modules/view/test/spec.coffee

```diff
@@ -120,4 +120,67 @@ describe 'hx-view', ->
     spy.should.have.been.called.with(2, 1)
     spy.should.have.been.called.with(3, 2)
     spy.should.have.been.called.with(4, 3)
-    hx.consoleWarning.should.not.have.been.called()
\ No newline at end of file
+    hx.consoleWarning.should.not.have.been.called()
+
+  it 'should only select direct children', ->
+    selection = hx.detached('div')
+
+    view = selection.view('div')
+      .enter(-> @append('div').add(hx.detached('span')).add(hx.detached('div')).node())
+      .update((d, e) -> hx.select(e).select('span').text(d))
+
+    view.apply(['a','b','c'])
+    selection.selectAll('div').text().should.eql(['a','','b','','c',''])
+    view.apply(['a','b','c'])
+    selection.selectAll('div').text().should.eql(['a','','b','','c',''])
```

Comment:
Added a 'simple' version of the error as well as a replication of the error from the issue